packages/synapse-interface/components/Activity/AirdropRewards.tsx

```diff
@@ -0,0 +1,410 @@
+      </div>
+    </Link>
+  )
+}
+
+const AirdropTxHeader = () => {
+  return (
+    <div className="grid grid-cols-3 border-none text-primary">
+      <div className="ml-1 text-greenText">ARB</div>
+      <div>Value</div>
+      <div className="mr-4 text-right">Tx Hash</div>
+    </div>
+  )
+}
+
+const RewardsTitle = ({ icon }) => {
+  return (
+    <div id="rewards-title" className="flex items-center space-x-1.5 p-3">
+      <Image
+        src={icon}
+        alt="reward chain icon"
+        className="w-4 h-4 mt-px rounded-full"
+      />
+      <div className="text-md">Rewards</div>
+    </div>
+  )
+}
+
+const RewardsAmountDisplay = ({
+  symbol,
+  tokenAmount,
+  dollarAmount,
+}: {
+  symbol: string
+  tokenAmount: string
+  dollarAmount: string
+}) => {
+  return (
+    <div
+      id="rewards-amount-display"
+      className="flex flex-wrap items-center space-x-1.5 leading-none"
+    >
+      <div className="text-white text-md">+${dollarAmount}</div>
+      <div className="mt-px text-sm text-secondary">
+        ({tokenAmount} {symbol})
+      </div>
+    </div>
+  )
+}
+
+export const HoverContentIcon = ({ children }) => {
+  const [isHovered, setIsHovered] = useState<boolean>(false)
+  return (
+    <div
+      id="hover-content-icon"
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+      className="relative"
+    >
+      <QuestionMarkCircleIcon
+        className="w-5 h-5 fill-white stroke-[#0A381B]"
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+      />
+      <HoverContent isHovered={isHovered}>{children}</HoverContent>
+    </div>
+  )
+}
+
+/** Helper Functions */
+const getArbStipRewards = async (connectedAddress: Address) => {
+  const { logs, data } = await getErc20TokenTransfers(
+    ARB.tokenAddress,
+    Rewarder.address,
+    connectedAddress,
+    ARB.network,
+    Rewarder.startBlock
+  )
+
+  const cumulativeRewards = calculateTotalTransferValue(data)
+
+  return {
+    logs: logs ?? [],
+    transactions: data,
+    cumulativeRewards,
+  }
+}
+
+const calculateTotalTransferValue = (data: any[]): bigint => {
+  let total: bigint = 0n
+  for (const item of data) {
+    if (item.transferValue) {
+      total += item.transferValue
+    }
+  }
+  return total
+}
+
+const parseTokenValue = (rawValue: bigint, tokenDecimals: number) => {
+  return trimTrailingZeroesAfterDecimal(
+    formatBigIntToString(rawValue, tokenDecimals, 3)
+  )
+}
+
+const convertTokensToDollarValue = (
+  tokenAmount: number | string,
+  tokenPrice: number | string
+) => {
+  return (Number(tokenAmount) * Number(tokenPrice)).toFixed(2)
+}
+
+const calculateDaysUntilStipEnds = () => {
+  const currentDate = new Date()
+  const targetDate = new Date('2024-03-30')
+
+  const timeDifference = Number(targetDate) - Number(currentDate)
+  const daysDifference = Math.floor(timeDifference / (1000 * 60 * 60 * 24)) + 1 // account for leap year
```

Comment:
`Math.floor + 1` is `Math.ceil`. Think this is less a leap year issue (`Date` handles that) and rounding + timezone related.